file/file_test.go

```diff
@@ -61,6 +63,51 @@ func TestNamedV0File(t *testing.T) {
 	}
 }
 
+func TestLargeFileReader(t *testing.T) {
+	buf := make([]byte, 1024*1024*1024)
+	ipfsutil.NewSeededRand(0xdeadbeef).Read(buf)
+	r := bytes.NewReader(buf)
+
+	ls := cidlink.DefaultLinkSystem()
+	storage := cidlink.Memory{}
+	ls.StorageReadOpener = storage.OpenRead
+	ls.StorageWriteOpener = storage.OpenWrite
+
+	f, _, err := builder.BuildUnixFSFile(r, "", &ls)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// get back the root node substrate from the link at the top of the builder.
+	fr, err := ls.Load(ipld.LinkContext{}, f, dagpb.Type.PBNode)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ufn, err := file.NewUnixFSFile(context.Background(), fr, &ls)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// read back out the file.
+	for i := 0; i < len(buf); i += 100 * 1024 * 1024 {
+		rs, err := ufn.AsLargeBytes()
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = rs.Seek(int64(i), io.SeekStart)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ob, err := io.ReadAll(rs)
```

Comment:
32-bit tests are failing on this:

```
runtime: out of memory: cannot allocate 209715200-byte block (3632005120 in use)
fatal error: out of memory
```

It's a big file, but `100 * 1024 * 1024` seeking from the start shouldn't be stressing it that much. Are we possibly hanging on to too much data to support the seek?